===================================================================
RCS file: /opt/sourcecast/data/ccvs/repository/gsl/vcl/unx/gtk/window/gtkframe.cxx,v
retrieving revision 1.19.2.4
retrieving revision 1.19.2.5
diff -u -r1.19.2.4 -r1.19.2.5
--- vcl/unx/gtk/window/gtkframe.cxx	2005/01/28 16:22:25	1.19.2.4
+++ vcl/unx/gtk/window/gtkframe.cxx	2005/02/02 10:07:05	1.19.2.5
@@ -66,6 +66,7 @@
 #include <keycodes.hxx>
 #include <wmadaptor.hxx>
 #include <salbmp.h>
+#include <salprn.h>
 #include <floatwin.hxx>
 #include <svapp.hxx>
 
@@ -1668,6 +1669,11 @@
             gtk_im_context_focus_out( pThis->m_pIMContext );
         }
     }
+
+    // ask for changed printers like generic implementation
+    if( pEvent->in )
+        if( static_cast< X11SalInstance* >(GetSalData()->pInstance_)->isPrinterInit() )
+            vcl_sal::PrinterUpdate::update();
 
     // FIXME: find out who the hell steals the focus from our frame
     // if we are plugged and a float opens; why does the same not
